### DISCUSSION TITLE
StatusOutput does not support set(long) when writing a string using the same connection that writes set of strings in transaction

### DISCUSSION THREAD

**Discussion by gigi888** [redis/lettuce#1896] [Q&A]
Hello,

Our AWS lambda uses a single connection to connect to redis 6 with lettuce 6.1.5. It writes a set of strings in transaction. 

                RedisCommands<String, String> redisCommands = redisConnection.sync();
                // cache the whole set in one transaction
                redisCommands.multi();
                val.stream().forEach(
                        (String s ) -> { redisCommands.sadd(key, s); }
                );
                redisCommands.expire(key, ttl);
                TransactionResult transactionResult = redisCommands.exec();
                if( transactionResult.wasDiscarded() ) {
                    logger.error(" Redis not able to write set of strings for key = " + key);
                }

In the meanwhile, another thread using the same connection, writes a simple string value to another key,

                RedisCommands<String, String> redisCommands = redisConnection.sync();
                redisCommands.set(key2, val);
                redisCommands.expire(key2, ttl);

while load testing this code, I got quite some
`java.lang.UnsupportedOperationException: io.lettuce.core.output.StatusOutput does not support set(long)`

here and there. I don't have such problem when only writing set. What is the suggested usage of our use case, when we need to write a set of strings with main thread, and just simple string with another thread?

Please help. Appreciate it

**Comment by mp911de** [comment]
When using transactions, then you must use a dedicated connection, see https://lettuce.io/core/release/reference/#_transactionsmulti.

**Reply by gigi888** [reply]
Thank you for the the answer!

**Comment by hisnamewasjaan** [comment]
If using synchronisation on the connection when it is being used in transactions, would this kind of error still be expected?

We are experiencing the same error using lettuce 6.2.6.RELEASE. 
We are using a single connection with all transactional code synchronised on the connection. And non-transactional use not synchronised.

Maybe it's just me, but this statement 

> You should not use transactional commands (MULTI) on shared connection.

from [Pipelining-and-command-flushing](https://github.com/lettuce-io/lettuce-core/wiki/Pipelining-and-command-flushing) is maybe a little ambiguous? What exactly does it mean?
Am I mistaken thinking that synchronising on the connection when using MULTI would be sufficient for it be considered not-shared?

Thanks for any clarification.
